elements/pf-progress/test/pf-progress.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { expect, html, fixture } from '@open-wc/testing';
+import { PfProgress } from '@patternfly/elements/pf-progress/pf-progress.js';
+
+describe('<pf-progress>', function() {
+  let element: PfProgress;
+
+  beforeEach(async function() {
+    element = await fixture<PfProgress>(html`
+      <pf-progress value="33"
+                  title="Progress title"
+                  label="Progress label">
+      </pf-progress>
+    `);
+  });
+
+  it('should upgrade', async function() {
+    const klass = customElements.get('pf-progress');
+    expect(element).to.be.an.instanceOf(klass).and.to.be.an.instanceOf(PfProgress);
+  });
+
+  it('should be accessible', async function() {
+    await expect(element).shadowDom.to.be.accessible();
+  });
+
+  it('should set the correct value on the progress bar', async function() {
+    const element = await fixture<PfProgress>(html`
+      <pf-progress value="33">
+      </pf-progress>
+    `);
+    expect(element.value).to.equal(33);
+  });
+
+  it('should set the correct title on the progress bar', async function() {
+    const element = await fixture<PfProgress>(html`
+      <pf-progress title="Progress title">
+      </pf-progress>
+    `);
+    expect(element.title).to.equal('Progress title');
+  });
+
+  it('should have the correct value with the max value set', async function() {
+    const max = Math.floor(Math.random() * 100);
+    const value = Math.floor(Math.random() * (max));
+
+    const element = await fixture<PfProgress>(html`
+      <pf-progress value="${value}" max="${max}">
+      </pf-progress>
+    `);
+    expect(element.value).to.equal(value);
+    expect(element.max).to.equal(max);
+  });
```

Comment:
```suggestion
  });
  
  describe('in a form', function() {
    let element: PfProgress;
    let form: HTMLFormElment;
    beforeEach(async function() {
      form = await createFixture(html`
        <form @submit=${e => e.preventDefault()}>
          <label for="progress">Progress</label>
          <pf-progress id="progress" name="progress" value="50"></pf-progress>
        </form>
      `);
      element = form.querySelector('pf-progress');
    });
    
    describe('submitting the form', function() {
      let submitEvent: SubmitEvent;
      beforeEach(async function() {
        setTimeout(() => form.requestSubmit());
        submitEvent = await oneEvent(form, 'submit');
      });
      it('sets the value on the form', function() {
        expect(form.elements.progress.value).to.equal(50);
      });
    });
  });
```

or some such. This will require some thought, since we coerce the min/value/max into a percentage.